### ISSUE TITLE
hive mind assimilation issue 

### ISSUE THREAD

**Issue by imsxz** [tgstation/tgstation#42066]
Not sure if this is intentional or not but it felt kinda funky. Assimilating has 2 periods of channeling which is alright. For the second channel, it only checks if the person is on screen once at the beginning of the second channel, and not the end. 

The reason it feels unintentional is because the second channel will begin even if the person has moved off screen, and fail to complete whether or not they re-appeared on screen. Likewise, if the target is on screen at the beginning of the second channel, it will complete successfully even if the person has feleported far away.

Related, and sorry if this was fixed earlier, but I’m just recalling from memory; you can start assimilating someone while they’re receiving a mindshield implant, and successfully complete it after the mindshield was inserted, allowing you to have a vessel that nobody will suspect.

tl;dr the beginning and ending of the second half of assimilation’s channeling needs more sanity checks
